### ISSUE TITLE
[php] PHP Intellisense Slow

### ISSUE THREAD

**Issue by TheColorRed** [Microsoft/vscode#1715]
Maybe it is just me, but Intellisense is very slow in php files. The result is that typing also gets slowed way down if Intellisense can not keep up.

On the same computer when I open up a Unity3d project typing and Intellisense work just fine, and I don't notice a slowdown.

**Comment by Ciantic** [comment]
But there is no intellisense for PHP in vscode? It's one of the most voted request though: https://visualstudio.uservoice.com/forums/293070-visual-studio-code/suggestions/7752606-php-intellisense

**Comment by osjn** [comment]
+1, the same question

**Comment by TheColorRed** [comment]
If you would prefer the term "autocomplete"...

**Comment by aeschli** [comment]
Autocomplete in PHP consists of all global symbols (functions, variables) as well as variables (starting with $) and functions defined in the code. 
Can you tell us more about when you see the slowness? Is it for large files?

**Comment by codetrash** [comment]
Im still having the same issue....its slow when typing php code (in PHP extension file)

**Comment by aeschli** [comment]
@codetrash Can you tell us more about when you see the slowness? Is it only for large files?
Can you maybe make a animated gif?

**Comment by maikwoehl** [comment]
[YouTube Video](https://youtu.be/c4EPXiDtsY8)

Sometimes it is very slow like this and the cursor freezes.

Before I capture it, it was even slower.

But after the capturing I closed the window and restarted VS Code.
And tried again to type. This time it was faster. Not as fast as in JS but usable.

Maybe VS Code builds a cache of the project or file and that takes some time.

I've also opened a file with 617 lines of code (a php class) and it was fast. I think it depends also on the scope/context.
